### ISSUE TITLE
Error: `TypeError: Cannot convert undefined or null to object` use `--report-needless-disables`

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1867]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

With CLI argument `--report-needless-disables`.

> Which rule, if any, is this issue related to?

does not matter

> What CSS is needed to reproduce this issue?

Need investigation

> What stylelint configuration is needed to reproduce this issue?

does not matter

> Which version of stylelint are you using?

e.g. `7.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS

> What did you expect to happen?

No warnings to be flagged

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
TypeError: Cannot convert undefined or null to object
    at /path/to/node_modules/stylelint/dist/needlessDisables.js:79:12
    at Array.forEach (native)
    at exports.default (/path/to/node_modules/stylelint/dist/needlessDisables.js:10:11)
    at prepareReturnValue (/path/to/node_modules/stylelint/dist/standalone.js:80:71)
    at /path/to/node_modules/stylelint/dist/standalone.js:102:16
```

**Comment by evilebottnawi** [comment]
Bugs here https://github.com/stylelint/stylelint/blob/master/src/needlessDisables.js#L8, some files can not have `disabledRanges` and `rangeData` is `undefined`.

**Comment by evilebottnawi** [comment]
`rangeData` is `undefined` only for ignored files https://github.com/stylelint/stylelint/blob/master/src/standalone.js#L140
